src/client/renderer/r_atlas.c

```diff
@@ -96,38 +96,60 @@ const r_atlas_image_t *R_GetAtlasImageFromAtlas(const r_atlas_t *atlas, const r_
 	return (const r_atlas_image_t *) g_hash_table_lookup(atlas->hash, image);
 }
 
-typedef struct {
-	uint16_t width, height;
-	uint16_t x, y;
+/**
+ * @brief Initialize an r_packer_t structure. If the packer is already
+ * created, clears the packer back to an initial state.
+ */
+void R_AtlasPacker_InitPacker(r_packer_t *packer, const r_pixel_t max_width, const r_pixel_t max_height, const r_pixel_t root_width, const r_pixel_t root_height, const uint32_t initial_size) {
```

Comment:
Also, I typically don't bother with `const` for non-pointer arguments. But I guess it can't hurt.